engine/src/module-engine.cpp

```diff
@@ -162,18 +164,30 @@ static Properties parse_property_name(MCStringRef p_name)
 	return P_CUSTOM;
 }
 
+static inline bool MCEngineEvalObjectOfScriptObject(MCScriptObjectRef p_object, MCObject *&r_object, uint32_t &r_part_id)
+{
+	__MCScriptObjectImpl *t_script_object_imp;
+	t_script_object_imp = (__MCScriptObjectImpl *)MCValueGetExtraBytesPtr(p_object);
+	if (t_script_object_imp -> handle == nil ||
+		!t_script_object_imp -> handle -> Exists())
+	{
+		// TODO: Throw script object doesn't exist error.
+		MCErrorCreateAndThrow(kMCGenericErrorTypeInfo, "reason", MCSTR("object does not exist"), nil);
```

Comment:
This should be abstracted to an kMCScriptObjectDoesNotExistError rather than using a generic.